src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
So after peeking at Chrome, there doesn't seem to be anything specific for string inlining:
https://github.com/v8/v8/blob/master/src/compiler/js-intrinsic-lowering.cc

Even their implementation of a ```Set``` would appear to be slower then going straight to asm.js (but I could be wrong):
https://github.com/v8/v8/blob/aca4a411e7082ffc790d79261ff383043de756ad/src/runtime/runtime-collections.cc